libs/experimental/langchain_experimental/llms/ollama_functions.py

```diff
@@ -340,14 +339,13 @@ def _generate(
 
         response_message_with_functions = AIMessage(
             content="",
-            additional_kwargs={
-                "function_call": {
-                    "name": called_tool_name,
-                    "arguments": json.dumps(called_tool_arguments)
-                    if called_tool_arguments
-                    else "",
-                },
-            },
+            tool_calls=[
+                ToolCall(
+                    name=called_tool_name,
+                    args=called_tool_arguments if called_tool_arguments else "",
```

Comment:
args is typed as a dict
```suggestion
                    args=called_tool_arguments if called_tool_arguments else {},
```